### ISSUE TITLE
TypeError: stat: can't specify None for path argument

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#866]
From a report on v0.3.0:

```
12:10:38 DEBUG    commands   command:run:503 Calling qutebrowser.mainwindow.statusbar.prompter.Prompter.prompt_accept(<qutebrowser.mainwindow.statusbar.prompter.Prompter busy=True loops=0 question=<qutebrowser.utils.usertypes.Question default='"$HOME/dl"       /' mode=<PromptMode.text: 2> text='Save file to:'> queue…)
[...]
12:10:38 DEBUG    downloads  downloads:fetch:759 fetch: PyQt5.QtCore.QUrl('http://img13.deviantart.net/3504/i/2013/326/8/0/comm__cas_by_ryuuen-d6v604g.png') -> comm__cas_by_ryuuen-d6v604g.png
12:10:38 DEBUG    downloads  downloads:set_filename:451 Setting filename to "$HOME/dl"       /outcast-angel
12:10:38 ERROR    misc       crashsignal:exception_hook:207 Uncaught exception
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/qutebrowser/browser/downloads.py", line 707, in <lambda>
    suggested_filename=suggested_fn))
  File "/usr/lib/python3.4/site-packages/qutebrowser/browser/downloads.py", line 732, in fetch_request
    suggested_filename)
  File "/usr/lib/python3.4/site-packages/qutebrowser/browser/downloads.py", line 779, in fetch
    download.set_filename(filename)
  File "/usr/lib/python3.4/site-packages/qutebrowser/browser/downloads.py", line 452, in set_filename
    if os.path.isfile(self._filename):
  File "/usr/lib/python3.4/genericpath.py", line 30, in isfile
    st = os.stat(path)
TypeError: stat: can't specify None for path argument
```

I wonder how that weird value of `"$HOME/dl"       /` even got in the config, but it seems like that's causing the issue.

**Comment by The-Compiler** [comment]
I can't reproduce this exact issue, but I can trigger other funny issues by setting `XDG_DOWNLOAD_DIR='"$HOME/dl"       /'` in `~/.config/user-dirs.dirs` - this is also the only way I can imagine that value could land there, as it's validated earlier for the config and the reporter didn't have a custom config.

**Comment by The-Compiler** [comment]
Assuming this was fixed in #1323.
